src/transformers/integrations.py

```diff
@@ -1067,15 +1096,20 @@ def _log_model_parameters(self, model):
             self._metadata_namespace["model_parameters"] = model.config.to_dict()
 
     def on_train_begin(self, args, state, control, model=None, **kwargs):
-        if self._should_reinitialize:
-            self._reinitialize_run()
-        self._should_reinitialize = True
+        self._ensure_run_with_monitoring()
+        self._force_reset_monitoring_run = True
```

Comment:
Could you remind me why do we have to force reseting run after training starts?

Reply:
We want the same number of experiments as training was run (we should remember that the user could provide the first run). Force at training start tells that -> create next time new experiment (even if there is one already created).